extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -315,22 +314,40 @@ export class ResourceTypePickerDialog extends DialogBase {
 				text: infoText.join(EOL)
 			};
 		}
+		if (!this.validateToolsEula()) {
+			this._dialogObject.okButton.label = loc.acceptEulaAndSelect;
+		}
 		this._toolsLoadingComponent.loading = false;
 	}
 
 	private validateToolsEula(): boolean {
-		const validationSucceeded = this._tools.every(tool => {
-			const eulaValidated = tool.validateEula();
+		this._eulaValidationSucceeded = this._tools.every(tool => {
+			const eulaValidated = tool.isEulaAccepted();
 			if (!eulaValidated) {
 				this._dialogObject.message = {
 					level: azdata.window.MessageLevel.Error,
-					text: tool.statusDescription!
+					text: [tool.statusDescription!, this._dialogObject.message.text].join(EOL)
 				};
 			}
 			return eulaValidated;
 		});
-		this._recheckEulaButton.hidden = validationSucceeded;
-		return validationSucceeded;
+		return this._eulaValidationSucceeded;
+	}
+
+	private async acquireEulaAndProceed(): Promise<boolean> {
+		this._dialogObject.message = { text: '' };
+		let eulaAccepted = true;
+		for (const tool of this._tools) {
+			eulaAccepted = tool.isEulaAccepted() || await tool.promptForEula();
+			if (!eulaAccepted) {
+				this._dialogObject.message = {
+					level: azdata.window.MessageLevel.Error,
+					text: [tool.statusDescription!, this._dialogObject.message.text].join(EOL)
+				};
+				break;
```

Comment:
Something to think about here is if we have multiple tools with eula's to accept. This is fine for now but might be worth coming back in after and redoing this slightly to build up the message for all failed checks.

Reply:
No appending is done here because it is not possible to accept EULAs of all tools at once. By design we prompt iteratively for each tools missing EULA. We do the following for each tool:
1. check for eula acceptatance for current tool and issue error for a tool missing EULA. Proceed to next tool if EULA alread accepted.
2. prompt to acquire EULA
3. proced to next tool if EULA accepted now else break out of all checking and issue error that EULA must be accepted and user has to try all over again (reentering this function) or cancel out of the workflow completely.

I have not tested with multiple tools requiring EULA acceptance but by code design that scenario should work fine. Note this is different from areToolsEulaAccepted where we will show information about missing eula for all tools.

Reply:
I missed that you're prompting for each tool in turn, yeah that should be good for now then. People generally shouldn't/won't be declining the EULA anyways so unlikely to be an actual issue regardless.